### PR TITLE
Correct Regex validation pattern for FeedTime

### LINKED ISSUE #248: [PE-D] Feed time allows invalid inputs
Feed time inputs when performing add, append and replace commands are allowed to be 2400, 2500, etc. till 2959.

For example when executing: add n/Lonesome George s/Galapagos Tortoise i/117 m/Flu f/2400

Error is only caught from values greater than 2960.

Perhaps valid input range for hour should be from 0 to 23 instead?

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: ameliatjy/ped#1

### PR DESCRIPTION
**Pull request by koonweee**
Resolve #248

### CODE REVIEW

**Comment by aizatazhar** [comment]
the updated regex seems to accept 3 digit numbers, e.g. `append 24211 259` or `append 24211 159`?


image

File: src/main/java/seedu/zookeep/model/feedtime/FeedTime.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
 public class FeedTime implements Comparable<FeedTime> {
     public static final String MESSAGE_CONSTRAINTS =
             "Feed times should be in the format HHmm, and it should not be blank";
-    public static final String VALIDATION_REGEX = "[0-2][0-9][0-5][0-9]";
+    public static final String VALIDATION_REGEX = "([01]?[0-9]|2[0-3])[0-5][0-9]";
```
Comment (aizatazhar) [inline src/main/java/seedu/zookeep/model/feedtime/FeedTime.java:13]: I think the `?` should be removed

i.e. `([01][0-9]|2[0-3])[0-5][0-9]`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -5,6 +5,18 @@ title: Developer Guide
 * Table of Contents
 {:toc}
 
+--------------------------------------------------------------------------------------------------------------------
```
Comment (aizatazhar) [inline docs/DeveloperGuide.md:8]: Sorry why does this pr modify the DG?

Comment (koonweee) [reply docs/DeveloperGuide.md:8]: Will close and redo PR due to unintended updates in master branch

**Pull request closed without merging**